### PR TITLE
Complete new implementation of the DocFxTocGenerator

### PR DESCRIPTION
**Pull request by mtirionMSFT**
This is a complete rewrite of the DocFxTocGenerator. By applying a new structure and newer techniques, the code is easier to understand and maintain. New options were added to configure the way the table of contents is generated.

We made sure that the default behavior of the tool is the same as the previous version, using the known flags. But as there were some edge-cases that were generated in the wrong way before, changes could be seen when using this new version. Some flags are still supported for backwards compatibility, but you are notified by the tool of this use and suggested to change to the new options.

The new options for generation are:
* Configuration when or how to generated index.md files
* Configuration what to use as reference for folders, where it can be omitted as well leaving just a navigation entry.
* Configuration how to order items in a folder.

The generation of multiple `toc.yml` files with the `-m | --multitoc` option is now implemented in a way that is understood by DocFx and also results in a better implementation in the website.

The new implementation also makes sure to not have orphaned markdown files or swagger files.

The README of the DocFxTocGenerator has been updated to reflect all these changes.

There is now also a test-project added to unit test all of the code, including the resolution of earlier reported issues.

### CODE REVIEW

File: src/DocFxTocGenerator/DocFxTocGenerator/Index/IndexService.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+/// <summary>
+/// Service to generate an index for a folder.
+/// </summary>
+public class IndexService
+{
+    private readonly LiquidService _liquidService;
+    private readonly IFileService _fileService;
+    private readonly ILogger _logger;
+
+    private readonly string _defaultIndexTemplate =
```
Comment (Ellerbach) [inline src/DocFxTocGenerator/DocFxTocGenerator/Index/IndexService.cs:20]: maybe add some comments on liquid template as not everyone is familiar with it

**Review by Ellerbach — COMMENTED** [review]
Amazing work! A deep refresh! Nice. Very few comments.

File: src/DocFxTocGenerator/DocFxTocGenerator.Test/ConfigFilesServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,176 @@
+﻿using Bogus;
```
[the remark this answers is not in the document]
Comment (mtirionMSFT) [reply src/DocFxTocGenerator/DocFxTocGenerator.Test/ConfigFilesServiceTests.cs:1]: Added

File: src/DocFxTocGenerator/DocFxTocGenerator/Actions/ContentInventoryAction.cs [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+public class ContentInventoryAction
+{
+    private readonly string _docsFolder;
+    private readonly bool _useOrder;
+    private readonly bool _useIgnore;
+    private readonly bool _useOverride;
+
+    private readonly IFileService? _fileService;
+    private readonly ConfigFilesService _configService;
+    private readonly ILogger? _logger;
```
[the remark this answers is not in the document]
Comment (mtirionMSFT) [reply src/DocFxTocGenerator/DocFxTocGenerator/Actions/ContentInventoryAction.cs:25]: Changed. Good one.

File: src/DocFxTocGenerator/DocFxTocGenerator/Actions/EnsureIndexAction.cs [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    /// <summary>
+    /// Run the action.
+    /// </summary>
+    /// <returns>0 on success, 1 on warning, 2 on error.</returns>
+    public Task<int> RunAsync()
+    {
+        int ret = 0;
+        _logger!.LogInformation($"\n*** ENSURE INDEX STAGE.");
```
[the remark this answers is not in the document]
Comment (mtirionMSFT) [reply src/DocFxTocGenerator/DocFxTocGenerator/Actions/EnsureIndexAction.cs:54]: Information - all shown when running in verbose mode. This shows the complete working of the tool. So, this is on purpose.

**Review by Ellerbach — APPROVED** [review]
Looks great! Thanks again!

File: src/DocFxTocGenerator/DocFxTocGenerator/Index/IndexService.cs (hunk 1 above)
Comment (mtirionMSFT) [reply src/DocFxTocGenerator/DocFxTocGenerator/Index/IndexService.cs:20]: Added some links in the LiquidService header. Other documentation is in the README.md.

**Pull request merged**